src/main/java/org/usfirst/frc/team5687/robot/subsystems/Intake.java

```diff
@@ -32,10 +32,50 @@ protected void initDefaultCommand() {
     }
 
     /**
+     * Checks if boulder is detected
+     *
+     * @return Whether or not the infrared sensor sees anything
+     */
+    public boolean isDetected() {
+        if (boulderSensor.getValue() < Constants.InfraRedConstants.DETECTION_THRESHOLD) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Checks if it is primed
+     * @return Whether or not the boulder is primed
+     */
+    public boolean isPrimeAble() {
+        return Math.abs(boulderSensor.getValue()- Constants.InfraRedConstants.PRIMED_OPTIMAL)< Constants.InfraRedConstants.PRIMED_TOLERANCE;
+    }
+
+    /**
+     * Checks if boulder is captured
+     * @return Whether or not the ball is captured
+     */
+    public boolean isCaptured(){
+         return ((Constants.InfraRedConstants.CAPTURED_OPTIMAL-boulderSensor.getValue()<Constants.InfraRedConstants.CAPTURED_TOLERANCE)||(boulderSensor.getValue()>Constants.InfraRedConstants.CAPTURED_OPTIMAL));
+    }
+    /**
      * Moves sensor data to smart dashboard
      */
     public void SendDashboardData() {
         SmartDashboard.putNumber("IR distance", boulderSensor.getValue());
+        if (!isDetected()){
```

Comment:
If conditions could be combined into else ifs, as they replace the same key `(Boulder is)` for SmartDashboard.